server/src/services/library.service.ts

```diff
@@ -366,258 +359,212 @@ export class LibraryService {
 
-    await this.jobRepository.queue({ name: JobName.METADATA_EXTRACTION, data: { id: assetId, source: 'upload' } });
+    await this.jobRepository.queue({ name: JobName.METADATA_EXTRACTION, data: { id: asset.id, source: 'upload' } });
 
-    if (assetType === AssetType.VIDEO) {
-      await this.jobRepository.queue({ name: JobName.VIDEO_CONVERSION, data: { id: assetId } });
+    if (asset.type === AssetType.VIDEO) {
+      await this.jobRepository.queue({ name: JobName.VIDEO_CONVERSION, data: { id: asset.id } });
     }
 
     return JobStatus.SUCCESS;
   }
 
-  async queueScan(id: string, dto: ScanLibraryDto) {
+  async queueScan(id: string) {
     await this.findOrFail(id);
 
     await this.jobRepository.queue({
-      name: JobName.LIBRARY_SCAN,
+      name: JobName.LIBRARY_QUEUE_SCAN,
       data: {
         id,
-        refreshModifiedFiles: dto.refreshModifiedFiles ?? false,
-        refreshAllFiles: dto.refreshAllFiles ?? false,
       },
     });
+    await this.jobRepository.queue({ name: JobName.LIBRARY_QUEUE_OFFLINE_CHECK, data: { id } });
   }
 
-  async queueRemoveOffline(id: string) {
+  async queueOfflineCheck(id: string) {
     this.logger.verbose(`Queueing offline file removal from library ${id}`);
-    await this.jobRepository.queue({ name: JobName.LIBRARY_REMOVE_OFFLINE, data: { id } });
+    await this.jobRepository.queue({ name: JobName.LIBRARY_OFFLINE_CHECK, data: { id } });
   }
 
-  async handleQueueAllScan(job: IBaseJob): Promise<JobStatus> {
-    this.logger.debug(`Refreshing all external libraries: force=${job.force}`);
+  async handleQueueAllScan(): Promise<JobStatus> {
+    this.logger.debug(`Refreshing all external libraries`);
 
     await this.jobRepository.queue({ name: JobName.LIBRARY_QUEUE_CLEANUP, data: {} });
 
     const libraries = await this.repository.getAll(true);
     await this.jobRepository.queueAll(
       libraries.map((library) => ({
-        name: JobName.LIBRARY_SCAN,
+        name: JobName.LIBRARY_QUEUE_SCAN,
+        data: {
+          id: library.id,
+        },
+      })),
+    );
+    await this.jobRepository.queueAll(
+      libraries.map((library) => ({
+        name: JobName.LIBRARY_QUEUE_OFFLINE_CHECK,
         data: {
           id: library.id,
-          refreshModifiedFiles: !job.force,
-          refreshAllFiles: job.force ?? false,
         },
       })),
     );
     return JobStatus.SUCCESS;
   }
 
-  async handleOfflineCheck(job: ILibraryOfflineJob): Promise<JobStatus> {
+  async handleAssetOfflineCheck(job: ILibraryOfflineJob): Promise<JobStatus> {
     const asset = await this.assetRepository.getById(job.id);
 
-    if (!asset) {
-      // Asset is no longer in the database, skip
+    if (!asset || asset.trashReason) {
+      // Skip if asset is missing or already trashed
+      // We don't want to trash an asset that has already been trashed because it can otherwise re-appear on the timeline if an asset is re-imported
```

Comment:
Not sure how I feel about this specifically, it leaves a bit of an open edge-case that won't be picked up until the next scan. Not sure how we'd resolve that really though while still using the trashReason stuff.

Reply:
The whole reason for the trashReason complexity is to handle this edge case. What edge case is left open?

Reply:
If an asset is trashed and goes offline what happens? We don't change the reason right? So if the user then untrashes the asset it could be in their timeline until the next scan but be broken. An edge case though and not something we can easily mitigate.

Reply:
Hah, I guess we could make another enum "trashed _and_ offline"